tests/test_cache.py

```diff
@@ -0,0 +1,82 @@
+import os
+import unittest
+
+from crashsimilarity.cache import TracesCache, DownloaderCache
+
+
+class CacheTest(unittest.TestCase):
+    default_cache_file_name = 'default.pickle'
+    other_file_name = 'other.pickle'
+
+    # don't want to use pyfake or mock os.system calls for such a small task
+    def setUp(self):
+        if os.path.exists(self.default_cache_file_name):
+            os.remove(self.default_cache_file_name)
+        if os.path.exists(self.other_file_name):
+            os.remove(self.other_file_name)
+
+    def tearDown(self):
+        self.setUp()
+
+
+class TestTracesCache(CacheTest):
+    lines = ['{"proto_signature": "a | CAPITAL_LETTERS | c", "uuid": "1", "signature": "c"}',
+             '{"proto_signature": "a | b | d", "uuid": "2", "signature": "d"}',
+             '{"proto_signature": "a | x | d", "uuid": "3", "signature": "d"}',
+             '{"proto_signature": "a | b | e", "uuid": "4", "signature": "same"}',
+             '{"proto_signature": "with | @0x end | drop", "uuid": "5", "signature": "drop"}',
+             '{"proto_signature": "with | xul.dll@", "uuid": "6", "signature": "ignored"}',
+             '{"proto_signature": "a | b | e", "uuid": "7", "signature": "same"}']
+    default_cache_file_name = 'traces_cache.pickle'
+
+    def test_traces_cache_build(self):
+        cache = TracesCache.build(self.lines)
+        expected = [(['a', 'capital_letters', 'c'], 'c', '1'),
+                    (['a', 'b', 'd'], 'd', '2'),
+                    (['a', 'x', 'd'], 'd', '3'),
+                    (['a', 'b', 'e'], 'same', '4'),
+                    (['with', '@0x', 'drop'], 'drop', '5')]
+        self.assertEqual(cache.traces, expected)
+        self.assertEqual(cache.name, 'traces')
+        self.assertEqual(cache.file_name, 'traces_cache.pickle')
+
+    def test_save_load(self):
+        cache = TracesCache.build(self.lines)
+        cache.dump(self.default_cache_file_name)
+        from_disk = TracesCache.load(self.default_cache_file_name)
+        self.assertEqual(from_disk.traces, cache.traces)
+        from_disk = TracesCache.load(self.other_file_name)
+        self.assertIsNone(from_disk)
+
+
+class TestDownloaderCache(CacheTest):
+    default_cache_file_name = 'downloader_cache.pickle'
+
+    def test_build(self):
+        cache = DownloaderCache.build()
+        self.assertDictEqual(cache, dict())
+        self.assertEqual(cache.name, 'downloader')
+        self.assertEqual(cache.file_name, self.default_cache_file_name)
+        cache = DownloaderCache()
+        self.assertDictEqual(cache, dict())
+        cache = DownloaderCache({'foo': 1, 42: 'bar'})
+        self.assertDictEqual(cache, {'foo': 1, 42: 'bar'})
+        cache = DownloaderCache.build({'foo': 1, 42: 'bar'}, file_name=self.other_file_name)
```

Comment:
You can remove the cache and recreate it here, so you don't need to use a second file.